### PR TITLE
Misc. fixes and SIGWINCH support.

### LINKED ISSUE #166: posix: Implement SIGWINCH
This allows programs like nano and vim to be resized.

### PR DESCRIPTION
**Pull request by qookei**
Fixes #166.

### CODE REVIEW

File: kernel/eir/generic/main.cpp [hunk 1]
```diff
@@ -104,37 +104,61 @@ address_t cutFromRegion(size_t size) {
...
-	
+
 	eir::panicLogger() << "Eir: Unable to cut memory from a region" << frg::endlog;
 	__builtin_unreachable();
 }
 
 void setupRegionStructs() {
-	for(size_t i = 0; i < numRegions; ++i) {
+	address_t buddyTreeAddrs[numRegions] = {};
+	address_t buddyTreeOverheads[numRegions] = {};
```
Comment (avdgrinten) [inline kernel/eir/generic/main.cpp:121]: Can't we just store this info within the `regions` array?

File: posix/subsystem/src/main.cpp [hunk 2]
```diff
@@ -1052,7 +1052,6 @@ async::result<void> serveRequests(std::shared_ptr<Process> self,
 
 			void *address;
 			if(req->flags() & MAP_ANONYMOUS) {
-				assert(req->fd() == -1);
```
Comment (avdgrinten) [inline posix/subsystem/src/main.cpp:1055]: Why is that removed? The assertion is broken, but shouldn't we return an error?

File: posix/subsystem/src/process.cpp [hunk 3]
```diff
@@ -466,6 +466,12 @@ struct CompileSignalInfo {
 	siginfo_t *si;
 };
 
+constexpr SignalDisposition defaultDispositions[65] = {
+	[SIGCHLD] = SignalDisposition::ignore,
+	[SIGURG] = SignalDisposition::ignore,
+	[SIGWINCH] = SignalDisposition::ignore
+};
```
Comment (avdgrinten) [inline posix/subsystem/src/process.cpp:473]: What about all other signals?

Comment (qookei) [reply posix/subsystem/src/process.cpp:473]: All other signals are zero initialized (`SignalDisposition::none`), and we treat `handler.disposition == none && defaultDispositions[sn] == none` as terminate.

File: posix/subsystem/src/main.cpp (hunk 2 above)
Comment (qookei) [reply posix/subsystem/src/main.cpp:1055]: As per the Linux man page, fd is ignored if MAP_ANONYMOUS is specified:
```
       MAP_ANONYMOUS
              The mapping is not backed by any file; its contents are initial‐
              ized to zero.  The fd argument is ignored; ...
```

File: posix/subsystem/src/process.cpp (hunk 3 above)
Comment (ArsenArsen) [reply posix/subsystem/src/process.cpp:473]: Other than TSTP, STOP and CONT, whose use-case we don't support yet

Comment (avdgrinten) [reply posix/subsystem/src/process.cpp:473]: Hm, one issue with this implementation is that there are default behaviors that that differ from all possible user specified behavior (for example for SIGCHLD, SIGSTOP, etc.). Maybe this is better expressed by a switch in the handler function?

**Comment by Dennisbonke** [comment]
What is the status of this PR?

**Review by avdgrinten — APPROVED** [review]
LGTM.

**Pull request merged**